charts/datadog/templates/_container-agent.yaml

```diff
@@ -59,6 +59,16 @@
     {{- include "containers-common-env" . | nindent 4 }}
     {{- include "fips-envvar" . | nindent 4 }}
     {{- include "processes-common-envs" . | nindent 4 }}
+    {{- if eq (include "should-enable-otel-agent" .) "true" }}
+    {{- if eq (include "agentIpcPort-envvar-agent-present" .) "false" }}
```

Comment:
usually we don't need to check if the user set an envvar because we add additional user envvars at the end of the list and so they override other envvars with the same key name.

Reply:
Good to know, this simplifies the PR a lot. Addressed in https://github.com/DataDog/helm-charts/pull/1661/commits/72eb8feeb23014f29adf9f7d34e13885af5fb104